src/NATS.Client.Core/NatsSubBase.cs

```diff
@@ -193,6 +195,20 @@ public virtual async ValueTask ReceiveAsync(string subject, string? replyTo, Rea
     {
         ResetIdleTimeout();
 
+        // check for empty payload conditions
+        if (payloadBuffer.Length == 0)
+        {
+            switch (Opts)
+            {
+            case { ThrowIfNoResponders: true } when headersBuffer is { Length: >= 12 } && headersBuffer.Value.Slice(8, 4).ToArray().SequenceEqual(NoRespondersHeaderSequence):
```

Comment:
The header code checking here is a little raw.  In the future, we might want to consider moving header parsing up the stack so that we have an actual `NatsHeaders` object here.

Reply:
`SequenceEqual` should work on span, no?

Reply:
What if it's multi-span?  Or is it guaranteed to never be, so could just check for this on the first span?

Reply:
ah... `ToSpan` instead of `ToArray` should address - made that change